File: SCALA/Cap08/c08ex03.scala
// c08ex03.scala

import scala.io.StdIn._

object Main {
  def main(args: Array[String]): Unit = {
    var numeros = new Array[Int](5)
    var i, j, x: Int = 0

    println("CLASSIFICACAO DE NUMEROS (CRESCENTE)\n")

    for (i <- 0 to 4) {
      print(s"${ i + 1 }o. numero: ")
      numeros(i) = readInt()
    }

    for (i <- 0 to 3) {
      for (j <- i + 1 to 4) {
        if (numeros(i) > numeros(j)) {
          x = numeros(i)
          numeros(i) = numeros(j)
          numeros(j) = x
        }
      }
    }

    println()
    for (i <- 0 to 4) {
      println(s"${ i + 1 }o. numero: ${ numeros(i) }")
    }

    print("\nTecle <Enter> para encerrar... ")
    readLine()
  }
}

File: DART/Cap08/c08ex03.dart
// c08ex03.dart

import 'dart:io';

void main() {
  List<int> numeros = List.filled(5, 0);
  int x;

  print("CLASSIFICACAO DE NUMEROS (CRESCENTE)\n");

  for (int i = 0; i <= 4; i++) {
    stdout.write("${i + 1}o. numero: ");
    numeros[i] = int.tryParse(stdin.readLineSync() ?? '') ?? 0;
  }

  for (int i = 0; i <= 3; i++)
    for (int j = i + 1; j <= 4; j++)
      if (numeros[i] > numeros[j]) {
        x = numeros[i];
        numeros[i] = numeros[j];
        numeros[j] = x;
      }

  print("");
  for (int i = 0; i <= 4; i++)
    print("${i + 1}o. numero: ${numeros[i]}");

  stdout.write("\nTecle <Enter> para encerrar... ");
  stdin.readLineSync();
}
